ecosystem/typescript/sdk/src/plugins/coin_client.ts

```diff
@@ -129,10 +130,11 @@ export class CoinClient {
       // The coin type to use, defaults to 0x1::aptos_coin::AptosCoin.
       // If you want to check the balance of a fungible asset, set this param to be the
       // fungible asset address
-      coinType?: string;
+      coinType?: MaybeHexString;
     },
   ): Promise<bigint> {
-    if (extraArgs?.coinType && AccountAddress.isValid(extraArgs.coinType)) {
+    const isTypeTag = (extraArgs?.coinType ?? "").toString().includes("::");
+    if (extraArgs?.coinType && !isTypeTag && AccountAddress.isValid(extraArgs.coinType)) {
```

Comment:
It's possible I made a wrong assumption that both functions checkBalance and coinTransfer can handle both v1 and v2 coins (CoinType and addresses) but that is why this exists. It's confusing because the string option can be for a CoinType aka "0x1::coin_type::CoinType" or a loosely typed hex string for the address of a fungible asset object address

Reply:
Ah, I see V2 type coins, gotta test that out then